colorizer_data/utils.py

```diff
@@ -522,3 +522,22 @@ def cast_feature_to_info_type(
             info.type, info.get_name()
         )
     )
+
+
+T = TypeVar("T", bound=Dict)
+
+
+def merge_dictionaries(a: T, b: T) -> T:
+    """Recursively merges key-value pairs of `b` into `a`, ignoring keys with `None` values."""
+    # This is basically a replacement for `{...a, ...b}` in JavaScript
+    if b == None or a == None:
+        return a
+    # Make shallow copy of a
+    a = {**a}
+
+    for key, value in b.items():
+        if isinstance(value, dict):
+            a[key] = merge_dictionaries(a[key], value)
```

Comment:
Would it be safer to use `a[key] = merge_dictionaries(a.get(key), value)`to avoid a KeyError in this case?